src/servo/connection.clj

```
@@ -68,10 +69,12 @@
 
 (defn ensure-table
   [{:keys [^Connection connection db-name]} table-name]
-  (let [table-name (->rt-name table-name)]
-    (when-not ((set (first (-> r (.db db-name) .tableList (.run connection))))
-               table-name)
-      (-> r (.db db-name) (.tableCreate table-name) (.run connection)))))
+  (locking connection
+    (let [table-name (->rt-name table-name)]
+      (when-not ((set (first (-> r (.db db-name) .tableList (.run connection)))) table-name)
+        (try
+          (-> r (.db db-name) (.tableCreate table-name) (.run connection))
+          (catch ReqlOpFailedError _ nil))))))
```

Are we sure swallowing the exception is correct here?

---

I was also wondering that.